### ISSUE TITLE
checkBox does not work in RTL languages, when you try to check the checkbox it does not check

### ISSUE THREAD

**Issue by jonasbernardo** [singerdmx/flutter-quill#2085]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Flutter Quill version

10.1.2

### Steps to reproduce

checkBox does not work in RTL languages, when you try to check the checkbox it does not check

### Expected results

checkBox does not work in RTL languages, when you try to check the checkbox it does not check

### Actual results

checkBox does not work in RTL languages, when you try to check the checkbox it does not check

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Additional Context

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by jonasbernardo** [comment]
https://github.com/user-attachments/assets/a9297c03-e730-4d99-b429-97ad45a6bce5

**Comment by jonasbernardo** [comment]
@CatHood0 
take a look at the video above

**Comment by jonasbernardo** [comment]
I looked in the flutter inspector, and look how the checkboxpoint class is, I don't know how to fix this, can anyone help me?
@CatHood0 
![Captura de tela de 2024-08-02 12-50-37](https://github.com/user-attachments/assets/301fe667-0f99-430f-b60a-ed1c0b0fc6a2)

**Comment by jonasbernardo** [comment]
@AtlasAutocode 
@singerdmx

**Comment by CatHood0** [comment]
I'll take a look. Give me some time. The issue is related with the harcoded widgets on the editor.

**Comment by jonasbernardo** [comment]
right, it's because after the update the checkboxes for RTL languages ​​became unusable

**Comment by CatHood0** [comment]
I notice the issue is more related from the Rendering. I still trying to fix it.

**Comment by jonasbernardo** [comment]
How is the progress going? Would it be possible to revert this change so that this function is not unusable in RTL?

**Comment by CatHood0** [comment]
I apologize, time is what I have had the least of so far. I will take the time to be able to solve this since I have to understand how the editor renders the checkbox and solve the root problem.

**Comment by CatHood0** [comment]
**Update**: This error seems to be more of a Flutter issue because the objects we render go to the right place, but it seems that widgets like GestureDetector, or InkWell no matter how much we force them, will not be able to move from the left.

However, I'm still trying to figure out how to solve this.

**Comment by CatHood0** [comment]
I'm still unable to fix this issue. The `InkWell` of the check can be moved even if you specify the Offset where it should be displayed. I think could be a issue from Flutter. When i have more time i try to fix again.

**Comment by jonasbernardo** [comment]
It would be good to revert the changes you made, because the checkboxes in rtl languages no longer work

**Comment by CatHood0** [comment]
@EchoEllet what do you think? should we revert this by now?

**Comment by EchoEllet** [comment]
Yes, we should revert any related change that caused the bug after the review.

#2060 Fix RTL issues, so it's supposed to fix this issue and related? The branch name called fix ltr (opposite of rtl)?

**Comment by jonasbernardo** [comment]
I evaluate it better, it really got better with the change, because before the enumerated lists were in ltr even though they were in an rtl language, it fixed a lot of things but the checkbox was still missing.

**Comment by jonasbernardo** [comment]
thinking better it is better with this change you made than the previous one, do not do the reversal, I did some tests here

**Comment by EchoEllet** [comment]
Have not tested it yet, if this is the case then we should not revert, instead improve the fix.

**Comment by CatHood0** [comment]
>  The branch name called fix ltr (opposite of rtl)?

You shouldn't pay attention to the names of the branches I create, because I name them first and then analyze the root of the problem that the branch is going to solve.

Take an example in [Feat: support for dynamic placeholders](https://github.com/singerdmx/flutter-quill/pull/2238) the branch is called `improve_headers` because at first I was only going to focus on that, but then I ended up noticing that a better implementation could be created and it became a name that has nothing to do with its main objective

**Comment by CatHood0** [comment]
This issue could not be fixed because `InkWell` or a `GestureDetector` cannot be moved even by forcing them. I don't know why these widgets cannot be moved using the `offset` into `text_line` (`RenderEditableTextLine` draws the leading of the lines)

**Comment by EchoEllet** [comment]
> You shouldn't pay attention to the names of the branches I create, because I name them first and then analyze the root of the problem that the branch is going to solve.

I assumed I misunderstood something since a fix regarding some RTL issues had already made, there was a minor issue that caused on LTR that was fixed after that.

> should we revert https://github.com/singerdmx/flutter-quill/pull/2060 by now?

I'm still not quite sure why you suggested reverting your fix when your fix is not a regression that caused this issue.

**Comment by CatHood0** [comment]
> I'm still not quite sure why you suggested reverting your fix when your fix is not a regression that caused this issue.

I thought it deserved a revert since it affects users to a certain extent (not too much). But now I think more calmly, it's true, it's a minor problem that only affects under certain conditions.

**Comment by jonasbernardo** [comment]
I was trying to find the problem, the code gives an offset in the paint child, but this offset is not working to change the position of the clickable area, maybe there is a way to give an offset to the clickable area

  if (textDirection == TextDirection.ltr) {
        final parentData = _leading!.parentData as BoxPar
        final effectiveOffset = offset + parentData.offset;
        context.paintChild(_leading!, effectiveOffset);
      } else {
        final parentData = _leading!.parentData as BoxParentData;
        final effectiveOffset = offset + parentData.offset;
        context.paintChild(
          _leading!,
          Offset(
            size.width - _leading!.size.width,
            effectiveOffset.dy,
          ),
        );
      }

![essa](https://github.com/user-attachments/assets/151d231e-b848-4bde-80ed-dd39b3ccfca6)
entData;

**Comment by jonasbernardo** [comment]
@CatHood0

**Comment by jonasbernardo** [comment]
I think this video can help?: https://youtu.be/svb41OLzCDY?si=Dg7EHrHZdj6IximQ&t=1516

**Comment by jonasbernardo** [comment]
To change the clickable area, I did a test and I seem to be able to find where to offset the clickable area
Above the paint method inside class RenderEditableTextLine.class I put this and it seems to have changed, now I need to be able to calculate the offset correctly within this method

  @override
  bool hitTest(BoxHitTestResult result, {required Offset position}) {
    // TODO: implement hitTest

    final attrs = line.style.attributes;
    final attribute =
        attrs[Attribute.list.key] ?? attrs[Attribute.codeBlock.key];
    //aqui
    final isCheck =
        attribute == Attribute.checked || attribute == Attribute.unchecked;

    if (isCheck) {
      return super.hitTest(result, position: Offset(300, 300));
    } else {
      return super.hitTest(result, position: position);
    }
  }

**Comment by jonasbernardo** [comment]
@CatHood0 
@singerdmx 
@EchoEllet 
I managed to change the position of the clickable area, now I just need to be able to calculate the correct horizontal offset, here is the method ready  
the -90 in the offset made the clickable area more to the right

Above the paint method inside class RenderEditableTextLine.class

@override
  bool hitTest(BoxHitTestResult result, {required Offset position}) {
    // TODO: implement hitTest

    final attrs = line.style.attributes;
    final attribute =
        attrs[Attribute.list.key] ?? attrs[Attribute.codeBlock.key];
    //aqui
    final isCheck =
        attribute == Attribute.checked || attribute == Attribute.unchecked;

    if (isCheck) {
      final parentData = _leading!.parentData as BoxParentData;
      final effectiveOffset = position + parentData.offset;
      return super.hitTest(result,
          position: Offset(effectiveOffset.dx - 90, effectiveOffset.dy));
    } else {
      return super.hitTest(result, position: position);
    }
  }

**Comment by jonasbernardo** [comment]
@CatHood0
@singerdmx
@EchoEllet
I managed to change the position of the clickable area, now I just need to be able to calculate the correct horizontal offset, here is the method ready
the -90 in the offset made the clickable area more to the right

Above the paint method inside class RenderEditableTextLine.class
 @override
  bool hitTest(BoxHitTestResult result, {required Offset position}) {
    // TODO: implement hitTest

    final attrs = line.style.attributes;
    final attribute =
        attrs[Attribute.list.key] ?? attrs[Attribute.codeBlock.key];
    //aqui
    final isCheck =
        attribute == Attribute.checked || attribute == Attribute.unchecked;

    if (isCheck) {
      final parentData = _leading!.parentData as BoxParentData;
      final effectiveOffset = position + parentData.offset;
      return super.hitTest(result,
          position: Offset(effectiveOffset.dx - 90, effectiveOffset.dy));
    } else {
      return super.hitTest(result, position: position);
    }
  }

**Comment by jonasbernardo** [comment]
@CatHood0
@singerdmx
@EchoEllet
I managed to change the position of the clickable area, now I just need to be able to calculate the correct horizontal offset, here is the method ready
the -90 in the offset made the clickable area more to the right

Above the paint method inside class RenderEditableTextLine.class
'@override'
bool hitTest(BoxHitTestResult result, {required Offset position}) {
final attrs = line.style.attributes;
final attribute =
    attrs[Attribute.list.key] ?? attrs[Attribute.codeBlock.key];
//aqui
final isCheck =
    attribute == Attribute.checked || attribute == Attribute.unchecked;

if (isCheck) {
  final parentData = _leading!.parentData as BoxParentData;
  final effectiveOffset = position + parentData.offset;
  return super.hitTest(result,
      position: Offset(effectiveOffset.dx - 90, effectiveOffset.dy));
} else {
  return super.hitTest(result, position: position);
}

}

**Comment by CatHood0** [comment]
Thanks for taking the efforts and the time. I'm unable to do something by some personal reasons. Do you will fix it? If not, give some time and i will fix it (using the new info. Thanks for it)

**Comment by CatHood0** [comment]
@jonasbernardo Can you show me your results? I mean a video showing how the position of the clickable area changes when you make your changes.

I ask you this because I already tested these changes (I made some modifications to be able to put it on the right based on the operating system's text direction), and I couldn't notice any relevant change, beyond that now when you hover over the checkbox it doesn't change (which normally doesn't happen).

Let's keep in mind that when we override the `hitTest` of `RenderEditableTextLine` we are applying it to the ENTIRE line. That is, we could be making a mistake in which area we should touch. Maybe we should look at `hitTestChildren` which does the same thing.

**Comment by jonasbernardo** [comment]
here is the video, depending on the offset value, the clickable area changes
 if (isCheck) {
    final parentData = _leading!.parentData as BoxParentData;
    final effectiveOffset = position + parentData.offset;
    return super.hitTest(result,
        position: Offset(effectiveOffset.dx - 90, effectiveOffset.dy));
  } else {
    return super.hitTest(result, position: position);
  }

https://github.com/user-attachments/assets/ec77e8f9-1976-493b-b441-3858dde2f360

**Comment by jonasbernardo** [comment]
for example if I put -100, it moves more to the right

**Comment by jonasbernardo** [comment]
@CatHood0

**Comment by CatHood0** [comment]
We need to take care more about this. Adding `hitTest` to the `RenderEditableTextLine` could cause more issues. We need to test this on all platforms. I saw your example works on android as expected, but, for desktop could have some unexpected behaviors. Let me take some time to debug it. Thanks for provide the example video.

**Comment by jonasbernardo** [comment]
Ok, If you send me the ready code I can test it on Android and IOS

**Comment by jonasbernardo** [comment]
any news?

**Comment by jonasbernardo** [comment]
@CatHood0  With your tip I managed to solve it, here is the code that solves it

just replace the hitTestChildren of the TextLine class with this code below

  @override
  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
    if (_leading != null) {
      final childParentData = _leading!.parentData as BoxParentData;
      Offset jonasOffset;
      if (textDirection == TextDirection.ltr) {
        jonasOffset = childParentData.offset;
      } else {
        jonasOffset = Offset(
            size.width - _leading!.size.width, childParentData.offset.dy);
      }
      final isHit = result.addWithPaintOffset(
        offset: jonasOffset,
        position: position,
        hitTest: (result, transformed) {
          assert(transformed == position - jonasOffset);
          return _leading!.hitTest(result, position: transformed);
        },
      );
      if (isHit) return true;
    }
    if (_body == null) return false;
    final parentData = _body!.parentData as BoxParentData;
    return result.addWithPaintOffset(
      offset: parentData.offset,
      position: position,
      hitTest: (result, position) {
        return _body!.hitTest(result, position: position);
      },
    );
  }

**Comment by jonasbernardo** [comment]
@EchoEllet

**Comment by jonasbernardo** [comment]
Another adjustment I made is in case the device's language is not supported by the device, quill does not give an error, returning the language to English if it does not find any    

FlutterQuillLocalizations lookupFlutterQuillLocalizations(Locale locale) 
I put the default at the end
default:
      return FlutterQuillLocalizationsEn();

and in 
extension LocalizationsExt on BuildContext 

 FlutterQuillLocalizations get loc {
    return FlutterQuillLocalizations.of(this) ?? FlutterQuillLocalizationsEn();
  }

I ask that anyone who can send these changes to the repository

**Comment by EchoEllet** [comment]
> FlutterQuillLocalizations.of(this) ?? FlutterQuillLocalizationsEn()

The class `FlutterQuillLocalizationsEn` is generated and we're not supposed to provide a default since if `FlutterQuillLocalizations.of(this)` returns null, it means the localization delegates are not set in the `MaterialApp` (or `WidgetsApp`) which is a indicator of error.

In Flutter v11 and above, it's required to set those manually where previously it provides them if they are not set.

**Comment by CatHood0** [comment]
@jonasbernardo i'll try your code and make the PR if it works as expected.
